src/main/java/net/glowstone/linkstone/Linkstone.java

```diff
@@ -1,16 +1,71 @@
 package net.glowstone.linkstone;
 
-import com.flowpowered.network.Message;
-import net.glowstone.net.message.play.game.PingMessage;
-import net.minecraft.server.Packet;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
 
 public class Linkstone {
-    public Linkstone() {
-        // TODO: ASM class trickery
+    public static String[] args;
+
+	public Linkstone() {
     }
 
-    public static Message packetToMessage(Packet message) {
-        String className = message.getClass().getName();
-        return new PingMessage(0);
+    public static void main(String[] arguments) throws MalformedURLException, IOException {
+    	args = arguments;
+
+    	System.out.println("Welcome to Linkstone! Bridging the gap between CraftBukkit and Glowstone");
+
+    	// TODO: Better configuration
+    	File configdir = new File("config");
+    	File config = new File(configdir, "linkstone.yml");
+    	configdir.mkdirs();
+    	if (!config.exists()) {
+    	    config.createNewFile();
+    	    List<String> lines = Arrays.asList("# Name for the glowstone jar", "glowstone: glowstone.jar");
+    	    Files.write(config.toPath(), lines, Charset.forName("UTF-8"));
+    	}
+
+    	String glowstone = "glowstone.jar";
+        for (String s : Files.readAllLines(config.toPath())) {
+            if (s.startsWith("glowstone:")) glowstone = s.substring("glowstone:".length()).trim();
+        }
+
+    	File glowjar = new File(glowstone);
+    	addURL(glowjar.toURI().toURL());
+    }
+
+    private static void addURL(URL u) throws IOException {
+        System.out.println("Adding glowstone jar to classpath ...");
+        URLClassLoader sysloader = (URLClassLoader) ClassLoader.getSystemClassLoader();
```

Comment:
The SystemClassLoader is no longer a URLClassLoader in java 9+. So this breaks compatibility with those versions.

Reply:
Once we migrated the code to my linkstone framework, this code will be dropped.